src/components/popover/widgets/capture-keystroke/capture-keystroke-button.component.ts

```diff
@@ -1,23 +1,77 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, EventEmitter, HostListener, Output } from '@angular/core';
+import { CaptureService } from '../../../../services/capture.service';
 
 @Component({
     selector: 'capture-keystroke-button',
     template: require('./capture-keystroke-button.component.html'),
     styles: [require('./capture-keystroke-button.component.scss')]
 })
-export class CaptureKeystrokeButtonComponent implements OnInit {
+export class CaptureKeystrokeButtonComponent {
+    @Output() capture = new EventEmitter<any>();
+
     private record: boolean;
+    private first: boolean;
+
+    constructor(private captureService: CaptureService) {
+        this.record = false;
+        this.captureService.initModifiers();
+        this.captureService.populateMapping();
+    }
 
-    constructor() { }
+    @HostListener('keyup')
+    onKeyUp() {
+        if (this.record && !this.first) {
+            this.saveScanCode();
+        }
+    }
+
+    @HostListener('keydown', ['$event'])
+    onKeyDown(e: KeyboardEvent) {
+        const code: number = e.keyCode;
 
-    ngOnInit() { }
+        if (this.record) {
+            this.first = false;
+
+            if (this.captureService.hasMap(code)) {
+                this.saveScanCode(this.captureService.getMap(code));
+            } else {
+                this.captureService.setModifier((e.location === 1), code);
+            }
+        } else if (code === 13) {
+            this.record = true;
+            this.first = true;
+        }
+    }
+
+    @HostListener('focusout')
+    onFocusOut() {
+        this.record = false;
+        this.reset();
+    }
 
     start(): void {
-        this.record = true;
+        if (!this.record) {
```

Comment:
I think you misunderstood me. I don't say that it's value is a constant `true` or anything like that.
I say it doesn't matter if you check for false or not. It would have the same behavior if you would leave this function as it was.